### ISSUE TITLE
Check if a request was websocket in `onRequest` hook

### ISSUE THREAD

**Issue by a-a-GiTHuB-a-a** [fastify/fastify-websocket#159]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

I would like to see if a request was websocket in the `onRequest` hook, maybe through a `req.ws` boolean.

### Motivation

I have a chat which uses the full declaration syntax with the websocket. However, I want my request to log differently if the request was a websocket.

### Example

Maybe like this:
```node
fastify.addHook("onRequest", async (req, res) => {
  if (req.ws) { //this is if it's a websocket
    req.log.info({req}, "websocket tryna connect");
  } else {
    req.log.info({req}, "regular old request");
  }
});
```

**Comment by airhorns** [comment]
Are you hoping to do something differently with the request if it is a websocket upgrade request, or really just log something differently?

**Comment by a-a-GiTHuB-a-a** [comment]
Actually both! See, when I open a websocket, it throws an INSANE amount of logs, almost one every .1 seconds!
If I fix this issue, I'm really just hoping to log it differently, but right now it's _crucial_ as the websocket never connects and I can't find out why.
I have a link if there's some other way to debug this thing.

**Comment by Eomm** [comment]
Are you using a react app?
I face a similar issue when one component was connecting to the WebSocket at every component render.

**Comment by a-a-GiTHuB-a-a** [comment]
No, it's not React.
Do you want the link?

**Comment by airhorns** [comment]
I think it should be pretty easy to add something like this! Would you like to submit a PR? I would suggest adding an `onRequest` hook that checks if the `request.raw[kWs]` is present and sets a nice boolean property that you can then inspect in the route handler. I am not certain that'd work, it might have to be a `preValidation` hook, but there's definitely enough information to know if the request is upgraded yet or not.

**Comment by a-a-GiTHuB-a-a** [comment]
Does it work with both the `websocket: true` and the `wsHandler` cases?

**Comment by airhorns** [comment]
Not sure -- shouldn't be too hard to write a test for though!

**Comment by climba03003** [comment]
Closed by #163
